custom/icds/translations/integrations/transifex.py

```diff
@@ -26,14 +28,21 @@ def __init__(self, domain, app_id, source_lang, project_slug, version=None, lang
         """
         if version:
             version = int(version)
-        key_lang = "en"  # the lang in which the string keys are, should be english
+        self.version = version
+        self.key_lang = "en"  # the lang in which the string keys are, should be english
+        self.lang_prefix = lang_prefix
         self.project_slug = project_slug
         self.is_source_file = is_source_file
         self.source_lang = source_lang
-        self.po_file_generator = POFileGenerator(domain, app_id, version, key_lang, source_lang, lang_prefix,
+        self.po_file_generator = POFileGenerator(domain, app_id, version, self.key_lang, source_lang, lang_prefix,
                                                  exclude_if_default)
+        self.parser = TranslationsParser(self)
```

Comment:
This dependency is a little weird - `Transifex` has an attribute called `parser`, which itself has `transifex` as an attribute.
It looks like `transifex.parser` is only called once, for `generate_excel_file`.  An alternative approach would be to make a `generate_excel_file` method on the `Transifex` class like this:
```python
def generate_excel_file(self):
    parser = TranslationsParser(self)
    return parser.generate_excel_file()
```
That way you could change the parser around without needing to worry about the calling code, since it doesn't access it directly.  Further, my preference would be to pass only the dependencies that `TranslationsParser` needs, rather than giving it this whole class.  That would avoid such tight coupling between the two classes.

Reply:
that makes sense.. that circular reference was not a good plan. Updating.